source/main_cr.f90

```diff
@@ -1290,13 +1277,8 @@ end subroutine envarsv
           pstop (j) = .false.
 
 !         values related to momentum
-!         old proton only terms:
-!          ejv   (j) = sqrt(ejfv(j)**2+pma**2)
-!          dpsv  (j) = (ejfv(j)-e0f)/e0f
-!          oidpsv(j) = one/(one+dpsv(j))
-
-          ejv   (j)   = sqrt(ejfv(j)**2+nucm(j)**2)              ! hiSix
-          dpsv  (j)   = (ejfv(j)*(nucm0/nucm(j))-e0f)/e0f         ! hiSix
+          ejv   (j)   = sqrt(ejfv(j)**2+nucm(j)**2)
+          dpsv  (j)   = (ejfv(j)*(nucm0/nucm(j))-e0f)/e0f
           oidpsv(j)   = one/(one+dpsv(j))
```

Comment:
I think that here we should put something like (maybe right after `mtc(j) assignment`):
```
if ( abs(nucm(j)/nucm0-one).lt.c1m15) then
   nucm(j)=nucm0
   if (nzz(j).eq.zz0 .or. naa(j).eq.aa0) then
     naa(j)=aa0
     nzz(j)=zz0
     mtc(j)=one
   else
     write(lout"(a)") "DIST> ERROR mass and/or charge mismatch wrt sync particle"
     call prror(-1)
   end if
end if
```